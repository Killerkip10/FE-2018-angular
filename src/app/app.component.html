<app-header></app-header>

<mat-drawer-container class="container">
  <mat-drawer *ngIf="tokenService.getToken()" mode="side" opened>
    <mat-nav-list>
      <a mat-list-item [routerLink]="['user-profile']" routerLinkActive="active">Profile</a>
      <a mat-list-item [routerLink]="['user-list']" routerLinkActive="active">Users list</a>

    </mat-nav-list>
  </mat-drawer>

  <mat-drawer-content>
    <router-outlet></router-outlet>
  </mat-drawer-content>
</mat-drawer-container>

<app-footer></app-footer>
