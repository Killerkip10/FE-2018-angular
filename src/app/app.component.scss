app-header{
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 2;
}
.container {
  margin-top: 56px;
  min-height: calc(100vh - 112px);
  mat-drawer{
    position: fixed;
    top: 56px;
    bottom: inherit;
  }
  .active {
    background-color: rgba(0,0,0,.04);
  }
}
