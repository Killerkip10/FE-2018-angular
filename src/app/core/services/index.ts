export * from './auth.service';
export * from './profile.service';
export * from './token.service';
export * from './search-user.service';
