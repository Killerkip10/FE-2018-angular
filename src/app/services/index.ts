export * from './profile.service';
export * from './token.service';
