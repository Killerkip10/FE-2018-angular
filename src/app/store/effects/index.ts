import {ProfileEffects} from './profile';
import {UsersEffects} from './users';

export const effects = [
  ProfileEffects,
  UsersEffects
];
