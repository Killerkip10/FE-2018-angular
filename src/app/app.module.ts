import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {RouterModule} from '@angular/router';
import {TranslateModule} from '@ngx-translate/core';

import {CoreModule} from './core/core.module';
import {AppRouting} from './app.routing';
import {AppComponent} from './app.component';
import {AppTranslate} from './app.translate';
import {
  MatSidenavModule,
  MatListModule
} from '@angular/material';
import {
  LoginPageModule,
  UserProfilePageModule,
  ForgotPasswordPageModule,
  SearchUsersPageModule
} from './pages';


@NgModule({
  imports: [
    CoreModule,
    CommonModule,
    RouterModule,
    TranslateModule,

    LoginPageModule,
    UserProfilePageModule,
    ForgotPasswordPageModule,
    SearchUsersPageModule,

    MatSidenavModule,
    MatListModule,

    AppTranslate,
    AppRouting
  ],
  declarations: [
    AppComponent
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
