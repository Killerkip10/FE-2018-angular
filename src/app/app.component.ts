import {Component, OnInit} from '@angular/core';
import {TranslateService} from '@ngx-translate/core';

import {Store} from '@ngrx/store';
import {GetProfile} from './store/actions/profile';

import {TokenService} from './core/services';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit {
  public tokenService = TokenService;

  constructor(
    translateService: TranslateService,
    private store: Store<null>
  ) {
    translateService.addLangs(['en', 'ru']);
    translateService.setDefaultLang('en');
  }

  ngOnInit(): void {
    this.store.dispatch(new GetProfile());
  }
}
