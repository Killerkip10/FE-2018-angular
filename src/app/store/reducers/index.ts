import {ActionReducerMap} from '@ngrx/store';

import * as fromUsers from './users';
import * as fromProfile from './profile';

export interface State {
  users: fromUsers.State;
  profile: fromProfile.State;
}

export const reducers: ActionReducerMap<State> = {
  users: fromUsers.reducer,
  profile: fromProfile.reducer
};
