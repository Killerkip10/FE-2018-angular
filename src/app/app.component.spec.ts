import {TestBed, async, ComponentFixture} from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {APP_BASE_HREF, CommonModule} from '@angular/common';
import {HttpClientModule} from '@angular/common/http';
import {RouterModule} from '@angular/router';
import {TranslateModule} from '@ngx-translate/core';

import {StoreModule} from '@ngrx/store';
import {reducers} from './store/reducers';
import {EffectsModule} from '@ngrx/effects';
import {effects} from './store/effects';

import { AppComponent } from './app.component';
import {AppTranslate} from './app.translate';
import {AppRouting} from './app.routing';
import {TokenService} from './core/services';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        HttpClientModule,
        CommonModule,
        RouterModule,
        TranslateModule,

        StoreModule.forRoot(reducers),
        EffectsModule.forRoot(effects),

        AppTranslate,
        AppRouting
      ],
      declarations: [
        AppComponent
      ],
      providers: [
        {provide: APP_BASE_HREF, useValue: '/'}
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', async(() => {
    expect(component).toBeTruthy();
  }));

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    class Token extends TokenService {
      static getToken(): string | null {
        return 'token';
      }
    }

    component.tokenService = Token;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    class Token extends TokenService {
      static getToken(): string | null {
        return null;
      }
    }

    component.tokenService = Token;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
