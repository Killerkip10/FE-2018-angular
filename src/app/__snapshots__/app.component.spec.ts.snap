// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snapshot 1`] = `
<app-root
  store={[Function Store]}
  tokenService={[Function Function]}
>
  <app-header /><mat-drawer-container
    class="container"
  >
    
    <mat-drawer-content>
      <router-outlet />
    </mat-drawer-content>
  </mat-drawer-container><app-footer />
</app-root>
`;

exports[`AppComponent snapshot 2`] = `
<app-root
  store={[Function Store]}
  tokenService={[Function Function]}
>
  <app-header /><mat-drawer-container
    class="container"
  >
    
    <mat-drawer
      mode="side"
      opened=""
    >
      <mat-list>
        <a
          href="/user-profile"
          mat-list-item=""
          ng-reflect-router-link="user-profile"
          ng-reflect-router-link-active="active"
          routerlinkactive="active"
        >
          
        </a>
        <a
          href="/search-users"
          mat-list-item=""
          ng-reflect-router-link="search-users"
          ng-reflect-router-link-active="active"
          routerlinkactive="active"
        >
          
        </a>
      </mat-list>
    </mat-drawer>
    <mat-drawer-content>
      <router-outlet />
    </mat-drawer-content>
  </mat-drawer-container><app-footer />
</app-root>
`;

exports[`AppComponent snapshot 3`] = `
<app-root
  store={[Function Store]}
  tokenService={[Function Function]}
>
  <app-header /><mat-drawer-container
    class="container"
  >
    
    <mat-drawer-content>
      <router-outlet />
    </mat-drawer-content>
  </mat-drawer-container><app-footer />
</app-root>
`;
